shop/src/main/java/ro/msg/learning/shop/strategy/SingleLocationSelectionStrategy.java

```diff
@@ -0,0 +1,75 @@
+package ro.msg.learning.shop.strategy;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import ro.msg.learning.shop.domain.Customer;
+import ro.msg.learning.shop.domain.Location;
+import ro.msg.learning.shop.domain.Order;
+import ro.msg.learning.shop.domain.Product;
+import ro.msg.learning.shop.dto.OrderCreateDto;
+import ro.msg.learning.shop.dto.ProductIdAndQuantityDto;
+import ro.msg.learning.shop.exception.ResourceNotFoundException;
+import ro.msg.learning.shop.mapper.OrderMapper;
+import ro.msg.learning.shop.service.CustomerService;
+import ro.msg.learning.shop.service.ProductService;
+import ro.msg.learning.shop.service.StockService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static ro.msg.learning.shop.service.OrderService.CUSTOMER_NOT_FOUND;
+import static ro.msg.learning.shop.service.OrderService.NO_LOCATION_WITH_SUFFICIENT_STOCK_FOUND;
+import static ro.msg.learning.shop.service.StockService.PRODUCT_NOT_FOUND_WITH_ID;
+
+
+public class SingleLocationSelectionStrategy implements LocationSelectionStrategy {
+
+    @Autowired
+    private StockService stockService;
+    @Autowired
+    private ProductService productService;
+    @Autowired
+    private CustomerService customerService;
+    @Autowired
+    private OrderMapper orderMapper;
+
+    @Override
+    public Order selectLocations(OrderCreateDto orderCreateDto) {
+        List<Location> locationsList = stockService.findLocationsByProducts(convertToProductList(orderCreateDto), convertToProductList(orderCreateDto).size());
+
+        Customer customer = customerService.findById(orderCreateDto.getIdCustomer()).orElseThrow(() -> new ResourceNotFoundException(CUSTOMER_NOT_FOUND));
+
+        if (!updateFirstLocationFoundWithSufficientStock(locationsList, orderCreateDto.getProductList())) {
+            throw new ResourceNotFoundException(NO_LOCATION_WITH_SUFFICIENT_STOCK_FOUND);
+        }
+
+        return orderMapper.orderCreateDtoToOrder(orderCreateDto, customer);
+
+    }
+
+    public boolean updateFirstLocationFoundWithSufficientStock(List<Location> locationsList, List<ProductIdAndQuantityDto> productIdAndQuantityDtoList) {
```

Comment:
returning boolean methods cannot be named "update", "do" etc. they have to be a question with yes/no answer
also, name needs to be shorter and less complex